internal/protocoltest/awsrestjson/api_op_JsonUnions.go

```diff
@@ -0,0 +1,74 @@
+// Code generated by smithy-go-codegen DO NOT EDIT.
+
+package awsrestjson
+
+import (
+	"context"
+	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/internal/protocoltest/awsrestjson/types"
+	smithy "github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+func (c *Client) JsonUnions(ctx context.Context, params *JsonUnionsInput, optFns ...func(*Options)) (*JsonUnionsOutput, error) {
+	stack := middleware.NewStack("JsonUnions", smithyhttp.NewStackRequest)
+	options := c.options.Copy()
+	for _, fn := range optFns {
+		fn(&options)
+	}
+	addawsRestjson1_serdeOpJsonUnionsMiddlewares(stack)
+	awsmiddleware.AddRequestInvocationIDMiddleware(stack)
+	smithyhttp.AddContentLengthMiddleware(stack)
+	AddResolveEndpointMiddleware(stack, options)
+	retry.AddRetryMiddlewares(stack, options)
+	awsmiddleware.AddAttemptClockSkewMiddleware(stack)
+	smithyhttp.AddErrorCloseResponseBodyMiddleware(stack)
+	smithyhttp.AddCloseResponseBodyMiddleware(stack)
+	stack.Initialize.Add(newServiceMetadataMiddleware_opJsonUnions(options.Region), middleware.Before)
+
+	for _, fn := range options.APIOptions {
+		if err := fn(stack); err != nil {
+			return nil, err
+		}
+	}
+	handler := middleware.DecorateHandler(smithyhttp.NewClientHandler(options.HTTPClient), stack)
+	result, metadata, err := handler.Handle(ctx, params)
+	if err != nil {
+		return nil, &smithy.OperationError{
+			ServiceID:     c.ServiceID(),
+			OperationName: "JsonUnions",
+			Err:           err,
+		}
+	}
+	out := result.(*JsonUnionsOutput)
+	out.ResultMetadata = metadata
+	return out, nil
+}
+
+type JsonUnionsInput struct {
+	Contents types.MyUnion
+}
+
+type JsonUnionsOutput struct {
+	Contents types.MyUnion
+
+	// Metadata pertaining to the operation's result.
+	ResultMetadata middleware.Metadata
+}
```

Comment:
Exported structs need to have documentation that structs and members with the name to avoid golint/vet failure.

Reply:
Sure, I can add docs in the protocol test pr